### ISSUE TITLE
zlib - modify cmake configuration to use standard system library

### ISSUE THREAD

**Issue by biocodz** [sortmerna/sortmerna#385]
Currently CMake configuration uses zlib library built on user machine in a custom location. Change this to use a standard library installation i.e. either pre-installed or installed using e.g. `apt install zlib1g zli1g-dev` on Debian flavour Linux. The custom build location should be optional.

**Comment by biocodz** [comment]
in progress. Will merge/close when done

**Comment by cmith44** [comment]
@biocodz This looks great! Making the building process simpler by using CMake presets also seems like a good idea. However it seems the preset performs the build in the home directory by default. Do you think it could be modified so that the default behaviour is to build in the sortmerna directory? Or maybe a new preset could be added for this, similarly to what we did in #375.

**Comment by biocodz** [comment]
Sure, the build location is easy to change. We can create all kinds of presets to accommodate any need. CMakePresets.json is reserved for most generic use cases, and CMakeUserPresets.json can always be used for customizations. Probably we don't need the env.jinja at all.

**Comment by cmith44** [comment]
Hi @biocodz, I tried running the above commands on a fresh environment (without RocksDB or zlib source files, with `zlib1g_dev` and `librocksdb-dev` libraries installed) and the building process stops and cannot find RocksDB config files. When I tried manually installing RocksDB using `python scripts/build.py -n sortmerna`, SortMeRNA compiled fine. Am I correct in understandingyou updated the building process to use zlib pre-built library but not RocksDB?

**Comment by biocodz** [comment]
Yeah, as can be seen in CMakePresets.json: "ROCKSDB_DIST": "$env{HOME}/a1/code/rocksdb/dist". 
My doubts about rocksdb are due the fact the standard Ubuntu installation offers a quite dated version e.g. on my machine:
```
lsb_release -a  -> Ubuntu 22.04.2 LTS
apt search rocksb
  librocksdb-dev/jammy 6.11.4-3 amd64
  librocksdb6.11/jammy 6.11.4-3 amd64
```
The latest version of sortmerna is built using v7.1.1, so the user would have to either build or to install rockdb from some other repos i.e. assumes a level of a developer knowledge. Consequently specifying a custom location would not be a problem. Using CMakeUserPresets.json is a standard CMake solution for custom settings.

Anyway, I'm open for suggestions on what default settings/locations to use

**Comment by cmith44** [comment]
>The latest version of sortmerna is built using v7.1.1

Is this a technical necessity due to a functionality that's available in RocksDB v7.1.1, or could SortmeRNA be built using v6.11.4? The reason this is important to us is because we'd like to implement some CI for SortMeRNA, and recompiling RocksDB represents a problem for CI workflows.

**Comment by biocodz** [comment]
I believe there was a reason for the upgrade but don't remember what was it and what version necessitated the upgrade.
Anyway, I'll try building and testing with a standard RocksDB installation even though I use Ubuntu 16 for preparing the releases, and the version of RocksDB there is even more ancient.

**Comment by biocodz** [comment]
Branch `build_3rdparty` implements new build strategy that takes care of all required 3rd party artifacts.
Major changes:
```
scripts/build.py => setup.py   # i.e. moved, renamed, modified
3rdparty.jinja                           # new 3rd party dependencies configuration file
cmake/presets/CMakePresets_rocksdb.json  # new file for building RocksDB
scripts/env.jinja                        # obsolete and will be removed
```
To build Sortmerna with the default settings (python >=3.6, cmake >= 3.21, C++17 capable toolchain are prerequisites):
```
git clone https://github.com/sortmerna/sortmerna.git
pushd sortmerna
git checkout build_3rdparty
python setup.py -n all  # default output path <sortmerna root>/dist/bin/sortmerna
# equivalently
python setup.py -n zlib
python setup.py -n rocksdb
python setup.py -n sortmerna

# test the build
python scripts/run.py -n t16 --smr-exe=dist/bin/sortmerna [--data-dir=../../data]
```

**Comment by biocodz** [comment]
fixed in release [4.3.7](https://github.com/sortmerna/sortmerna/releases/tag/v4.3.7)
